src/mbgl/renderer/layers/render_raster_layer.cpp

```diff
@@ -31,4 +36,35 @@ bool RenderRasterLayer::hasTransition() const {
     return unevaluated.hasTransition();
 }
 
+void RenderRasterLayer::uploadBuckets(gl::Context& context, RenderSource* source) {
+    if (renderTiles.size() > 0) {
```

Comment:
There is a higher likelihood for the source to be a tile-based `RasterSource`. This order reduces the need for `dynamic_cast` for the majority of cases when `renderTiles` are available.